### DISCUSSION TITLE
Configuration barcode generation

### DISCUSSION THREAD

**Discussion by BhaskarUday** [micjahn/ZXing.Net#306] [General]
My scanner device understand that given barcode is configuration barcode and not data barcode if we give "reader-init=true" parameter in bwipjs.  Please provide any pointers for achieving the same using ZXing.Net library

**Comment by micjahn** [comment]
I have no idea what you mean. ZXing.Net reads the barcode and gives you the content of it. It's the responsibility of the calling code to interpret the content in the correct way.

**Reply by BhaskarUday** [reply]
Please kindly review http://bwip-js.metafloor.com/demo/demo.html in java script.  There we have options as parsefnc=true.  When this is set if the barcode data is given as ^FNCBEPLVL3, the reader at my side understands that it is configuration barcode and configures the beeper level to 3.
Similarly any feasibility in ZXing.Net please or any other alternatives.

**Reply by prlwtz** [reply]
The Zxing library misses the feature to enable "Menu" barcodes. This is a lesser-known feature of the Aztec barcode specification.
It requires a certain bit (flag) to be set in the header of the barcode. Of the top of my head, it is the highest bit of the length field.

**Reply by prlwtz** [reply]
I had made enhancements to a local copy of the 0.16.4 version of the library to include this feature about 2 years ago. I can inject the same changes into the 0.16.6 version. It's not a massive change. If desired, I can send the changes to the author?

**Reply by micjahn** [reply]
You can send me the changes or better open a new pull request.

**Reply by prlwtz** [reply]
[ZXing-Aztec-Menu.zip](https://github.com/micjahn/ZXing.Net/files/5996132/ZXing-Aztec-Menu.zip)

Not sure how to open a pull request, unfortunately I have not done much with github yet. Attached the changes to the encoder. Also contains a screenshot how to use and showing the difference in barcode. Hope I amended the code along the style of the ZXing library.

**Reply by prlwtz** [reply]
I then realized that the ZXing decoder also does not parse Aztec barcodes with the menu flag set (because of the "invalid" length). I have amended the detector to cope with it and enhanced the ResultMetaData to indicate whether the decoded barcode had the menu flag set.
 
[ZXing-Aztec-Menu-EncodeDecode.zip](https://github.com/micjahn/ZXing.Net/files/5999247/ZXing-Aztec-Menu-EncodeDecode.zip)
